lib/ansible/modules/cloud/amazon/ec2_ami.py

```diff
@@ -428,221 +397,291 @@ def create_image(module, ec2):
     device_mapping = module.params.get('device_mapping')
     tags = module.params.get('tags')
     launch_permissions = module.params.get('launch_permissions')
+    image_location = module.params.get('image_location')
+    enhanced_networking = module.params.get('enhanced_networking')
+    billing_products = module.params.get('billing_products')
+    ramdisk_id = module.params.get('ramdisk_id')
+    sriov_net_support = module.params.get('sriov_net_support')
 
     try:
-        params = {'name': name,
-                  'description': description}
+        params = {
+            'Name': name,
+            'Description': description
+        }
 
-        images = ec2.get_all_images(filters={'name': name})
+        images = connection.describe_images(
+            Filters=[
+                {
+                    'Name': 'name',
+                    'Values': [name]
+                }
+            ]
+        ).get('Images')
 
+        # ensure that launch_permissions are up to date
         if images and images[0]:
-            # ensure that launch_permissions are up to date
-            update_image(module, ec2, images[0].id)
+            update_image(module, connection, images[0].get('ImageId'), resource)
```

Comment:
The current test failures are related to using update_image to set launch permissions.

The module exits with 'AMI not updated' when the launch permission doesn't change. Perhaps make set_launch_permissions a separate method to be used by both create_image and update_image.

Reply:
@willthames I got the tests working and found that it was requiring default images that weren't present. Have added these images for each region. 

There was also an error due to `.block_device_mapping['/dev/xvda']` being called on the result instead of `.block_device_mappings[0].ebs`. I've changed this in the tests to pass but not sure if this is correct or of backwards compatibility had been broken.


Thanks for the doco on setting up btw, I decided to turn your notes into a <a href="https://github.com/wilvk/vagrant-ansible-tests">Vagrant/Ansible provisioned box</a> to save going through the steps each time if that's ok.

Reply:
This seems as though backward compatibility has been broken - you might need to create a block_device_mapping dict based on the result of block_device_mappings.